packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.test.ts

```diff
@@ -208,10 +177,10 @@ describe('BalancesController', () => {
         },
       });
 
-    controller.start();
     mockListMultichainAccounts.mockReturnValue([mockBtcAccount]);
     messenger.publish('AccountsController:accountAdded', mockBtcAccount);
-    await controller.updateBalances();
+
+    await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
I understand that this "forces" the pending promises to be run (since we have some `async` calls happening during the `'AccountsController:accountAdded'` event.

But I think we should wrap this in a function to make it more explicit.

Also, I found this post: https://stackoverflow.com/a/51045733

So I think we should for something like:

```ts
async function waitForAllPromises(): Promise<void> {
  // Wait for next tick to flush all pending promises. It's requires since
  // we are testing some asynchronous execution flows that are started by
  // synchronous calls.
  await new Promise(process.nextTick);
}
```

And then use everywhere instead:
```suggestion
    await waitForAllPromises();
```

WDYT?